development/cpp/configuring_an_ide.rst

```diff
@@ -263,6 +263,60 @@ Test it:
 - It should break at the point!
 
 
+.. _doc_configuring_an_ide_vs:
+
+Visual Studio
+--------------
+
+Visual Studio Community is free for use and has lots of great features, such as memory view, performance view, source control and more.
+You can get it `at microsoft <https://visualstudio.microsoft.com/downloads/>`_.
+
+Setup
+^^^^^^
+
+In order to start developing with this IDE please follow:
+
+- Open Visual Studio Installer and install the C++ package to start working with C++
+
+.. image:: img/vs_1_install_cpp_package.png
+
+- Use cmd or powershell to configure the project to generate ``.sln`` files and stuff and run: ``scons p=windows vsproj=yes``
+
+- Now open the godot folder by clicking "Open a project or solution" and open ``godot.sln``
+
+You can now start developing with Visual Studio.
+
+Debugging
+^^^^^^^^^^
+
+Debugging with Visual Studio is easy. This allows the user to examin the source code of godot, stop at spesific points in the code,
+make changes, and view them on the run.
+
+.. Note:: Debugging the Godot Engine inside the editor will require an extra setup step.
+             
+             Because openning godot opens firstly the Project Manager, and not the project(game) you are working on,
+             the debugger will detach as soon as you open a project, meanning the debugger will stop, even though godot is still running.
+
+             The reason is because the debugger attaches to the Project Manager process, and when you open a project it will close and open
```

Comment:
```suggestion
             The reason is that the debugger attaches to the Project Manager process only. When you open a project, it will close and open another process for
```